src/client.cpp

```diff
@@ -65,6 +65,19 @@ RawVector fetch(CharacterVector server, CharacterVector method, RawVector reques
   // perform request
   grpc_op ops[6];
   grpc_op *op;
+
+  int METADATA_LENGTH = metadata.length() / 2;
+  grpc_metadata meta_c[METADATA_LENGTH];
+
+  if (METADATA_LENGTH > 0) {
```

Comment:
I think you don't need the if statement here, flow won't enter the loop when length is 0 anyway.